lib/http/node.js

```diff
@@ -76,15 +75,40 @@ AWS.NodeHttpClient = AWS.util.inherit({
 
   writeBody: function writeBody(stream, httpRequest) {
     var body = httpRequest.body;
-
-    if (body && WritableStream && ReadableStream) { // progress support
-      if (!(body instanceof Stream)) body = AWS.util.buffer.toStream(body);
-      body.pipe(this.progressStream(stream, httpRequest));
-    }
+    var totalBytes = parseInt(httpRequest.headers['Content-Length'], 10);
+    var loadedBytes = 0;
 
     if (body instanceof Stream) {
+      // for progress support of streaming content
+      // tap the data event of the stream in addition to piping
+      body.on('data', function(chunk) {
```

Comment:
Very good point. Thanks for taking the time to review in depth.

In one of my previous tests I saw that the pipe to the writable progressStream caused buffers to be copied. But now I'm not sure about it completely, as it might have mixed with the buffer path in the code in my runs. 

Anyhow, I think to change the path of stream-body progress to use Transform stream to emit the events in the familiar pattern `body.pipe(progressStream).pipe(stream)`. Transform stream will push the same buffer so will surely avoid copies. sounds right?